PCP/src/main/java/PCP/packets/PCPMsgUserToUserPacket.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * this is a school project under "The Unlicence".
+ */
+
+package PCP.packets;
+import PCP.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+/**
+ *
+ * @author gfurri20
+ * @author Jacopo_Wolf
+ * @author Alessio789
+ */
+public class PCPMsgUserToUserPacket implements IPCPpacket
+{
+    private byte[] senderId;
+    private String destinationAlias;
+    private String message;
+
+
+//<editor-fold defaultstate="collapsed" desc="getter and setters">
+    
+    public String getDestinationAlias()
+    {
+        return destinationAlias;
+    }
+    
+    public void setDestinationAlias( String destinationAlias )
+    {
+        this.destinationAlias = destinationAlias;
+    }
+    
+    public String getMessage()
+    {
+        return message;
+    }
+    
+    public void setMessage( String message )
+    {
+        this.message = message;
+    }
+    
+//</editor-fold>
+    
+    
+    public PCPMsgUserToUserPacket( byte[] senderId, String dstAlias, String msg )
+    {
+        this.senderId = senderId;
+        this.destinationAlias = dstAlias;
+        this.message = msg;
+    }
+        
+    @Override
+    public OpCode getOpCode()
+    {
+        return OpCode.MsgUserToUser;
+    }
+
+    @Override
+    public byte[] header()
+    {
+        byte[] buffer = new byte[this.size()];
+        int i = 0;
+        //Opcode
+        buffer[i++] = OpCode.MsgUserToUser.getByte();
+
+        //SenderID
+        for(byte b : senderId)
+            buffer[i++] = b;
+
+        return buffer;
+    }
+
+    @Override
+    public int size()
+    {
+        return 5 + destinationAlias.length() + message.length();
+    }
+
+    @Override
+    public Collection<byte[]> toBytes()
+    {
+        Collection<byte[]> out = new ArrayList<>();
+        
+        
+        byte[] messageB = message.getBytes( StandardCharsets.ISO_8859_1 );
+        
+        int NpacketsToSent = 
+                (
+                    this.message.length() / 
+                    (PCP.Min.MAX_PACKET_LENGHT - 5 - this.destinationAlias.length())
+                ) 
+                + 1 ;
+        
+        int messageRelativeMaxLenght = PCP.Min.MAX_PACKET_LENGHT - 5 - destinationAlias.length();
+        
+        
+        int messagePointer = 0;
+        for ( int packetN = 0; packetN < NpacketsToSent; packetN++ )
+        {
+        
+            byte[] buffer = this.header();
```

Comment:
this will inevitably generate an **IndexOutOfBounds** exception, because the buffer lenght is the header's.
Should be corrected in other files if the error is repeated

Reply:
No because header() returns a buffer which is initialize with the length take from size()